src/cmd/create.go

```diff
@@ -0,0 +1,490 @@
+
+	logrus.Infof("Used image will be: %s", fullImageName)
+
+	// Look for the toolbox image on local machine
+	imageFound := findLocalToolboxImage(imageName)
+
+	imagePulled := false
+	if !imageFound {
+		logrus.Infof("Image '%s' was not found on the host machine", imageName)
+		imagePulled, err := pullToolboxImage(fullImageName)
+		if !imagePulled {
+			if err != nil {
+				logrus.Fatalf("Failed to pull image: %v", err)
+			}
+			// If the image was not pulled but there is no error then the user said no to pulling an image
+			os.Exit(0)
+		}
+		logrus.Infof("Image '%s was pulled", fullImageName)
+	} else {
+		logrus.Infof("Image '%s' was found", fullImageName)
+	}
+
+	// Check if the image is a Toolbox image
+	// FIXME: In the future this check will have to be done only for local images because for pulled images it can be done by inspecting their manifest before pulling them
+	if imageFound || imagePulled {
+		if !createFlags.unsafe {
+			isToolboxImage, err := checkIfToolboxImage(imageName)
+			if !isToolboxImage {
+				logrus.Fatalf("Image '%s' is not a Toolbox image: %v", imageName, err)
+			}
+			logrus.Infof("Image '%s' is a Toolbox image", imageName)
+		} else {
+			logrus.Infof("Skipping check if image '%s' is a Toolbox image", imageName)
+		}
+	}
+
+	logrus.Info("Looking for group for sudo")
+	sudoGroup = utils.GetGroupForSudo()
+	if sudoGroup == "" {
+		logrus.Fatal("Group for sudo was not found")
+	}
+	logrus.Infof("Group for sudo is %s", sudoGroup)
+
+	logrus.Info("Getting user ID")
+	currentUser, err := user.Current()
+	if err != nil {
+		logrus.Fatal("Failed to get user information")
+	}
+	userID := currentUser.Uid
+	logrus.Infof("User ID is %s", userID)
+
+	// Start assembling the arguments for Podman
+	createArgs := []string{
+		"create",
+		"--dns", "none",
+		"--env", fmt.Sprintf("TOOLBOX_PATH=%s", viper.GetString("TOOLBOX_PATH")),
+		"--group-add", sudoGroup,
+		"--hostname", "toolbox",
+		"--ipc", "host",
+		"--label", "com.github.containers.toolbox=true",
+		"--name", containerName,
+		"--network", "host",
+		"--no-hosts",
+		"--pid", "host",
+		"--privileged",
+		"--security-opt", "label=disable",
+		"--userns=keep-id",
+		"--user", "root:root"}
+
+	command := []string{
+		"toolbox", "--log-level", "debug", "init-container",
+		"--home", viper.GetString("HOME"),
+		"--monitor-host",
+		"--shell", viper.GetString("SHELL"),
+		"--uid", userID,
+		"--user", viper.GetString("USER")}
+
+	logrus.Info("Checking if toolbox.sh profile exists")
+	if utils.PathExists("/etc/profile.d/toolbox.sh") {
+		logrus.Info("Found /etc/profile.d/toolbox.sh")
+
+		toolboxProfileBind = []string{"--volume", "/etc/profile.d/toolbox.sh:/etc/profile.d/toolbox.sh:ro"}
+		createArgs = append(createArgs, toolboxProfileBind...)
+	} else if utils.PathExists("/usr/share/profile.d/toolbox.sh") {
+		logrus.Info("Found /usr/share/profile.d/toolbox.sh")
+
+		toolboxProfileBind = []string{"--volume", "/usr/share/profile.d/toolbox.sh:/etc/profile.d/toolbox.sh:ro"}
```

Comment:
`toolboxProfileBind` doesn't get appended in this code path.

Reply:
Good catch!